### DISCUSSION TITLE
Ruby bin dir / path missing?

### DISCUSSION THREAD

**Discussion by galund** [Homebrew/discussions#13] [Everyday usage]
Brew on linux has installed ruby into 
`/home/linuxbrew/.linuxbrew/bin/ruby`

this is fine, and on my path (set up in .bash_profile)

But ruby executables themselves go in `/home/linuxbrew/.linuxbrew/lib/ruby/gems/2.7.0/bin/ `

This is not on my path.

I'm sticking the following into .bash_profile as a workaround:

```
export PATH="$(ruby -r rubygems -e 'puts Gem.default_bindir'):$PATH"
```

But is there a better way, and shouldn't installing ruby have done that for me somehow?

**Comment by jonchang** [comment]
This should be displayed as a Caveat when you install ruby. Something like this will appear:

```
% brew install ruby
[...]
==> Caveats
==> ruby
By default, binaries installed by gem will be placed into:
  /home/linuxbrew/.linuxbrew/lib/ruby/gems/2.7.0/bin

You may want to add this to your PATH.
```

**Reply by galund** [reply]
Thanks for the reply. That's very likely so, but that note was probably in middle of a large amount of other output, so it's easy to miss.

Personally I think the 'caveat' mechanism is a terrible user experience (i.e. "it's okay that X broke because somewhere sometime we told you about it once") but I have no suggestion for how to fix it. In the case of paths, maybe there could be a mechanism by which homebrew maintains a collection of path directories suggested by each package, so you only have to call one homebrew command from your profile in order for it to keep your PATH tidy. Sounds like a substantial feature request, I guess :-)

Thanks :-)
